docs/development/design/4-DN-Sequencing_pv_events_from_otel_unique_graphs/Algorithm_Overview.puml

```diff
@@ -0,0 +1,95 @@
+@startuml
+    start
+    :Input ->\nIterable of map of event_id to OTelEvent\nasync flag,\nmap of event type to list of lists of async following event types;
+    partition "sequence_otel_jobs" {
+        repeat
+            :Input ->\nmap of event id to OTelEvent,\nasync flag,\nmap of event type to list of lists of async following event types;
+            :Call "sequence_otel_event_job"|
+            :yield the iterable of PVEvents;
+        repeat while (for each map of event id to OTelEvent)
+    }
+    end
+    partition "sub algorithms" {
+        start
+        :Input ->\nmap of event id to OTelEvent\nasync flag,\nmap of event type to list of lists of async following event types;
+        group "sequence_otel_event_job"
+            :find root OTelEvent;
+            :Input ->\nroot OTelEvent,\nmap of event_id to OTelEvent,\nasync flag,\nmap of event type to list of lists of async following event types;
+            :Call "sequence_otel_event_and_ancestors";
+            repeat
+                :get OTelEvent from map of event id to OTelEvent;
+                :create PVEvent from OTelEvent using previous event ids;
+                :yield PVEvent;
+            repeat while (for each event id and previous event ids\nin map of event id to previous event ids)
+        end group
+        end
+        start
+        :Input ->\nOTelEvent,map of event id to OTelEvent,\nprevious event ids list,\nasync flag,\nmap of event type to list of lists of async following event types;
```

Comment:
Should this input be the same as above?
```suggestion
        group "sequence_otel_event_job"
            :find root OTelEvent;
            :Input ->\nroot OTelEvent,\nmap of event_id to OTelEvent,\nasync flag,\nmap of event type to list of lists of async following event types;
```

Reply:
No that previous event ids list is required